### ISSUE TITLE
Incorrect unused parentheses

### ISSUE THREAD

**Issue by i-am-the-slime** [intellij-purescript/intellij-purescript#371]
I will gather examples here:

<img width="774" alt="image" src="https://github.com/intellij-purescript/intellij-purescript/assets/1588055/bddd210f-9d93-4b44-8984-e653d8a4654f">
 - [ ] `either (\err -> { stdout:"", stderr: intercalateMap "\n" renderHumanError err }) identity`

**Comment by i-am-the-slime** [comment]
<img width="573" alt="image" src="https://github.com/intellij-purescript/intellij-purescript/assets/1588055/27708471-91aa-44df-a3b0-4258a6c2248f">

```
nes :: forall @a. IsSymbol a => MakeNonEmpty a => NonEmptyString
nes = NES.nes (Proxy :: Proxy a)
```

**Comment by i-am-the-slime** [comment]
```
            ((Array.drop 1 props.headings <#> Just) <> [ Nothing ])
```

**Comment by i-am-the-slime** [comment]
```
      (icon (if open then rxCross2 else rxRowSpacing) { size: "15px" })
```

Says the parens around (if then else) are unnecessary

**Comment by baffalop** [comment]
Yeah, so I've been encountering false positives for pretty much all lambda parameters.

Here's a simple one: ```2 `pow` (n - 1)```
